src/resolver.rs

```diff
@@ -85,8 +97,86 @@ impl Resolver {
         }
     }
 
+    pub fn query_family_multicast_groups(
+        &mut self,
+        handle: &GenetlinkHandle,
+        family_name: &'static str,
+    ) -> impl Future<Output = Result<HashMap<String, u32>, GenetlinkError>> + '_
+    {
+        if let Some(groups) = self.get_groups_cache_by_name(family_name) {
+            Either::Left(futures::future::ready(Ok(groups)))
+        } else {
+            let mut handle = handle.clone();
+            Either::Right(async move {
+                // Create the request message to get family details
+                let mut genlmsg: GenlMessage<GenlCtrl> =
+                    GenlMessage::from_payload(GenlCtrl {
+                        cmd: GenlCtrlCmd::GetFamily,
+                        nlas: vec![GenlCtrlAttrs::FamilyName(
```

Comment:
If this multicast group is also in `GetFamily` reply, I think we should merge this function into `query_family_id()`, so we have single netlink request to get both caches ready.